lightning/src/ln/msgs.rs

```diff
@@ -270,84 +215,64 @@ pub struct OpenChannelV2 {
-	pub delayed_payment_basepoint: PublicKey,
-	/// Used to derive an HTLC payment key to sender for transactions broadcast by counterparty
-	pub htlc_basepoint: PublicKey,
-	/// The first to-be-broadcast-by-sender transaction's per commitment point
-	pub first_per_commitment_point: PublicKey,
-	/// A request to pre-set the to-sender output's scriptPubkey for when we collaboratively close
-	pub shutdown_scriptpubkey: Option<ScriptBuf>,
-	/// The channel type that this channel will represent.
-	///
-	/// If this is `None`, we derive the channel type from the intersection of
-	/// our feature bits with our counterparty's feature bits from the [`Init`] message.
-	/// This is required to match the equivalent field in [`OpenChannel::channel_type`].
-	pub channel_type: Option<ChannelTypeFeatures>,
-	#[cfg(taproot)]
-	/// Next nonce the channel initiator should use to create a funding output signature against
-	pub next_local_nonce: Option<musig2::types::PublicNonce>,
 }
 
-/// An accept_channel2 message to be sent by or received from the channel accepter.
+/// An open_channel2 message to be sent by or received from the channel initiator.
 ///
 /// Used in V2 channel establishment
 ///
+// TODO(dual_funding): Add spec link for `open_channel2`.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct OpenChannelV2 {
+	/// Common fields of `open_channel(2)`-like messages
+	pub common_fields: CommonOpenChannelFields,
+	/// The feerate for the funding transaction set by the channel initiator
+	pub funding_feerate_sat_per_1000_weight: u32,
+	/// The locktime for the funding transaction
+	pub locktime: u32,
+	/// The second to-be-broadcast-by-channel-initiator transaction's per commitment point
+	pub second_per_commitment_point: PublicKey,
+	/// Optionally, a requirement that only confirmed inputs can be added
+	pub require_confirmed_inputs: Option<()>,
+}
+
+/// Contains fields that are both common to [`accept_channel`] and `accept_channel2` messages.
+///
+/// [`accept_channel`]: https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#the-accept_channel-message
 // TODO(dual_funding): Add spec link for `accept_channel2`.
 #[derive(Clone, Debug, Hash, PartialEq, Eq)]
-pub struct AcceptChannelV2 {
-	/// The same `temporary_channel_id` received from the initiator's `open_channel2` message.
+pub struct CommonAcceptChannelFields {
+	/// The same `temporary_channel_id` received from the initiator's `open_channel2` or `open_channel` message.
 	pub temporary_channel_id: ChannelId,
-	/// Part of the channel value contributed by the channel acceptor
-	pub funding_satoshis: u64,
 	/// The threshold below which outputs on transactions broadcast by the channel acceptor will be
 	/// omitted
 	pub dust_limit_satoshis: u64,
-	/// The maximum inbound HTLC value in flight towards channel acceptor, in milli-satoshi
+	/// The maximum inbound HTLC value in flight towards sender, in milli-satoshi
```

Comment:
I believe this was unintended.

```suggestion
	/// The maximum inbound HTLC value in flight towards channel acceptor, in milli-satoshi
```

Reply:
I believe `channel acceptor` and `channel initiator` were being used for the `OpenChannelV2` and `AcceptChannelV2` messages. So kept that for common fields. It's still true here and more specific, but `sender` is still 100% valid too!